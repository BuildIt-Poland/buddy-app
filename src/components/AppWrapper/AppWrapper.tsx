import React from 'react';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
import { Route, BrowserRouter } from 'react-router-dom';

import { ROUTES } from '../../shared/routes';
import TaskDetails from '../TaskDetails';
import Login from '../Login';
import TasksList from '../TasksList';
import AddNewbie from '../AddNewbie';
import NewbieSelect from '../NewbieSelect';
import ContactDetails from '../ContactDetails';
import AddTask from '../AddTask';
import ErrorPage from '../ErrorPage';

const useStyles = makeStyles(theme => ({
  container: {
    minWidth: theme.breakpoints.values['xs'],
    marginTop: theme.spacing(9),
  },
}));

const Root: React.FC = () => {
  const classes = useStyles();

  return (
    <Container
      data-testid={'root'}
      className={classes.container}
      component='main'
      maxWidth='md'>
      <BrowserRouter basename={ROUTES.BASE}>
        <Route path={ROUTES.ERROR} exact component={ErrorPage} />
        <Route path={ROUTES.LOGIN} exact component={Login} />
        <Route path={ROUTES.BUDDY_ADD_NEWBIE} exact component={AddNewbie} />
        <Route path={ROUTES.BUDDY_SELECT_NEWBIE} exact component={NewbieSelect} />
        <Route path={ROUTES.BUDDY_TASKS_LIST} exact component={TasksList} />
        <Route path={ROUTES.BUDDY_TASK_DETAILS} exact component={TaskDetails} />
        <Route path={ROUTES.BUDDY_NEWBIE_DETAILS} exact component={ContactDetails} />
        <Route path={ROUTES.NEWBIE_BUDDY_DETAILS} exact component={ContactDetails} />
        <Route path={ROUTES.BUDDY_ADD_TASK} exact component={AddTask} />
        <Route path={ROUTES.NEWBIE_TASKS_LIST} exact component={TasksList} />
        <Route path={ROUTES.NEWBIE_TASK_DETAILS} exact component={TaskDetails} />
      </BrowserRouter>
    </Container>
  );
};
export default Root;
