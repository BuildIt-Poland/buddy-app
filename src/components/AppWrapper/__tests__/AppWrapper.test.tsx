import React from 'react';
import { create } from 'react-test-renderer';

import AppWrapper from '../AppWrapper';

jest.mock('@material-ui/core/Container', () => 'Container');
jest.mock('../../Login', () => 'Login');
jest.mock('../../ProtectedRoute/ProtectedRoute', () => 'ProtectedRoute');
jest.mock('../../RouteRedirect/RouteRedirect', () => 'RouteRedirect');

describe('Component - AppWrapper', () => {
  test('renders correctly', () => {
    const component = create(<AppWrapper></AppWrapper>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
