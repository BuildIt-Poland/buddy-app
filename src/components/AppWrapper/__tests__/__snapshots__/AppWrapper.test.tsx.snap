// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AppWrapper renders correctly 1`] = `
<Container
  className="makeStyles-root-1"
  component="main"
  maxWidth="md"
/>
`;
